koku/masu/processor/tasks.py

```diff
@@ -915,3 +920,54 @@ def remove_stale_tenants():
                 KokuTenantMiddleware.tenant_cache.clear()
         for name in data:
             LOG.info(f"Deleted tenant: {name}")
+
+
+@celery_app.task(name="masu.processor.tasks.process_openshift_on_cloud", queue=GET_REPORT_FILES_QUEUE, bind=True)
+def process_openshift_on_cloud(self, schema_name, provider_uuid, bill_date, tracing_id=None):
+    """Process OpenShift on Cloud parquet files using Trino."""
+    provider = Provider.objects.get(uuid=provider_uuid)
+    provider_type = provider.type.replace("-local", "")
+
+    table_info = {
+        Provider.PROVIDER_AWS: {"table": "aws_line_items_daily", "date_columns": ["lineitem_usagestartdate"]},
+        Provider.PROVIDER_AZURE: {"table": "azure_line_items", "date_columns": ["usagedatetime", "date"]},
+        Provider.PROVIDER_GCP: {
+            "table": "gcp_line_items_daily",
+            "date_columns": ["usage_start_time", "usage_end_time"],
+        },
+    }
+    table_name = table_info.get(provider_type).get("table")
+    if isinstance(bill_date, str):
+        bill_date = ciso8601.parse_datetime(bill_date)
+    year = bill_date.strftime("%Y")
+    month = bill_date.strftime("%m")
+
+    # We do not have fine grain control over specific days in a month for
+    # OpenShift on Cloud parquet generation. This task will clear and reprocess
+    # the entire billing month passed in as a parameter.
+    where_clause = f" WHERE source='{provider_uuid}' AND year='{year}' AND month='{month}' "
+    table_count_sql = f"SELECT count(*) FROM {table_name}" + where_clause
+
+    count, _ = execute_trino_query(schema_name, table_count_sql)
+    count = count[0][0]
+
+    processor = OCPCloudParquetReportProcessor(
+        schema_name, "", provider_uuid, provider_type, 0, context={"tracing_id": tracing_id, "start_date": bill_date}
+    )
+    remove_files_not_in_set_from_s3_bucket(
+        tracing_id, processor.parquet_ocp_on_cloud_path_s3, 0, processor.error_context
+    )
+    for i, offset in enumerate(range(0, count, settings.PARQUET_PROCESSING_BATCH_SIZE)):
+        query_sql = (
+            f"SELECT * FROM {table_name}"
+            + where_clause
+            + f"OFFSET {offset} LIMIT {settings.PARQUET_PROCESSING_BATCH_SIZE}"
+        )
+        results, columns = execute_trino_query(schema_name, query_sql)
+        data_frame = pd.DataFrame(data=results, columns=columns)
+        for column in table_info.get(provider_type).get("date_columns"):
```

Comment:
If there is no `provider_type` key in `table_info` the second `get()` call will raise an an `AttributeError` on `None`.
```suggestion
        for column in table_info.get(provider_type, {}).get("date_columns"):
```

Reply:
True, true. I'm actually totally okay with an exception here. Never say never, but the API code is the only thing that will be calling this task and it will return a 400 if you provide an unacceptable provider / type. If we  add another OCP on Cloud type, I'd rather this get noisy and fail as a way of alerting us to to the need to add info on the specific fields for that type, rather than this silently pass and do nothing.